bankAccount-system/index.js

```diff
@@ -0,0 +1,150 @@
+class BankAccount {
+    constructor(accountNumber, firstName, lastName, balance, dailyWithdrawalLimit) {
+        this._accountNumber = accountNumber;
+        this._firstName = firstName;
+        this._lastName = lastName;
+        this._accountHolder = `${firstName} ${lastName}`;
+        this._balance = balance;
+        this._transactions = [];
+        this._dailyWithdrawalLimit = dailyWithdrawalLimit
+        this._withdrawnToday = 0;
+        this._lastWithdrawalDate = new Date().getDate();
+    }
+
+    get accountNumber() {
+        return this._accountNumber;
+    }
+
+    get firstName () {
+        return this._firstName;
+    }
+
+    get lastName () {
+        return this._lastName;
+    }
+
+    get accountHolder () {
+        return this._accountHolder;
+    }
+
+    get balance () {
+        return this._balance;
+    }
+
+    deposit(amount) {
+        if (amount <= 10) {
+            throw new Error('Invalid amount entered. Minimum depost is 10');
+        }
+        else {
+            this._balance += amount;
+            this._transactions.push(new Transaction('Deposit', amount));
+            return `${this._accountHolder}, ${amount} has been deposited to your account. Your new balance is ${this._balance}`
+        }
+    }
+    
+    withdraw(amount) {
```

Comment:
This implementation of withdrawal limit is lost to the answer but not still accurate. as the daily withdrawal limit takes effect par day, with the below implementation the limit tracking rolls over to the next day.